includes/Modules/Analytics_4/Settings.php

```diff
@@ -88,6 +94,11 @@ protected function get_sanitize_callback() {
 				if ( isset( $option['useSnippet'] ) ) {
 					$option['useSnippet'] = (bool) $option['useSnippet'];
 				}
+				if ( isset( $option['googleTagID'] ) && ! empty( $option['googleTagID'] ) ) {
+					if ( ! preg_match( '/^(G|GT|AW)-[a-zA-Z0-9]+$/', $option['googleTagID'] ) ) {
+						return false;
+					}
```

Comment:
This shouldn't return `false` here as the return value is the sanitized option as a whole. If invalid, it it should essentially apply the default instead. In other cases such as a boolean, we cast it to ensure the expected type is set.

Reply:
Apparently I misunderstood the IB. Changed it as proposed.